lib/db_utils.go

```diff
@@ -1106,7 +1113,7 @@ func DBSetWithTxn(txn *badger.Txn, snap *Snapshot, key []byte, value []byte, eve
 		ancestralValue, getError = DBGetWithTxn(txn, snap, key)
 
 		// If there is some error with the DB read, other than non-existent key, we return.
-		if getError != nil && getError != badger.ErrKeyNotFound {
+		if getError != nil && !errors.Is(getError, badger.ErrKeyNotFound) {
```

Comment:
Huh... do we need to do this in all places or were you just being extra defensive here?

Reply:
It's the new standard way of checking errors for a specific error type. I've been gradually fixing them all as I go.